### PR TITLE
Updating intrinsic sizing keywords data

### PR DESCRIPTION
**Pull request by rachelandrew**
Firefox 66 has unprefixed the min-content and max-content keywords: https://bugzilla.mozilla.org/show_bug.cgi?id=1322780. I've updated the compat data based on this and my testing. The fit-content keyword is still left prefixed.

The data is pretty patchy so I also tested on some other browsers to fill out some nulls, so have updated Edge and Safari too.

All of the sizing keywords were combined in some files, and as data typically is different, I split these out. I added sections for these keywords where there were none.

### CODE REVIEW

File: css/properties/block-size.json [hunk 1]
```diff
@@ -72,7 +72,165 @@
...
+              "version_added": false
+            },
+            "safari_ios": {
+              "version_added": false
+            },
+            "samsunginternet_android": {
+              "version_added": "5.0"
+            },
+            "webview_android": {
+              "version_added": "46"
```
Comment (ddbeck) [inline css/properties/block-size.json]: I think something's amiss here. This data has webview getting `max-content`—and the others below—before the parent `block-size` and before Chrome on desktop and Android, which strikes me as unlikely. Could you revisit this data?

**Comment by rachelandrew** [comment]
Good catch, I must have missed those when trying to split the data in a sensible manner. Fixed.

Also while just checking I hadn't been thinking of something else I got the Opera data so have included that, and done some other fixing up.

File: css/properties/max-height.json [hunk 2]
```diff
@@ -9,7 +9,7 @@
               "version_added": "1"
             },
             "chrome_android": {
-              "version_added": null
+              "version_added": true
```
Comment (jpmedley) [inline css/properties/max-height.json]: Should be 18.

File: css/properties/max-height.json [hunk 3]
```diff
@@ -162,6 +207,70 @@
...
+                  {
+                    "version_added": "46"
+                  },
+                  {
+                    "prefix": "-webkit-",
+                    "version_added": "22"
+                  }
+                ],
+                "chrome_android": {
+                  "version_added": "46"
```
Comment (jpmedley) [inline css/properties/max-height.json]: This landed with a -webkit- prefix in 25.

File: css/properties/max-height.json [hunk 4]
```diff
@@ -162,6 +207,70 @@
...
+                  }
+                ],
+                "safari_ios": {
+                  "version_added": null
+                },
+                "samsunginternet_android": {
+                  "version_added": "5.0"
+                },
+                "webview_android": {
+                  "version_added": "46"
```
Comment (jpmedley) [inline css/properties/max-height.json]: Webkit prefix should be true.

File: css/properties/max-width.json [hunk 5]
```diff
@@ -53,52 +53,137 @@
...
+                {
+                  "version_added": "46"
+                },
+                {
+                  "prefix": "-webkit-",
+                  "version_added": "22"
+                }
+              ],
+              "chrome_android": {
+                "version_added": "46"
```
Comment (jpmedley) [inline css/properties/max-width.json]: webkit = 25

File: css/properties/max-width.json [hunk 6]
```diff
@@ -53,52 +53,137 @@
...
               "safari_ios": {
                 "version_added": null
               },
               "samsunginternet_android": {
-                "version_added": false
+                "version_added": "5.0"
               },
               "webview_android": {
-                "version_added": null
+                "version_added": "46"
```
Comment (jpmedley) [inline css/properties/max-width.json]: webkit = true

File: css/properties/max-width.json [hunk 7]
```diff
@@ -160,6 +245,71 @@
...
+                  {
+                    "version_added": "46"
+                  },
+                  {
+                    "prefix": "-webkit-",
+                    "version_added": "22"
+                  }
+                ],
+                "chrome_android": {
+                  "version_added": "46"
```
Comment (jpmedley) [inline css/properties/max-width.json]: webkit = 25

File: css/properties/max-width.json [hunk 8]
```diff
@@ -160,6 +245,71 @@
...
+                  }
+                ],
+                "safari_ios": {
+                  "version_added": null
+                },
+                "samsunginternet_android": {
+                  "version_added": "5.0"
+                },
+                "webview_android": {
+                  "version_added": "46"
```
Comment (jpmedley) [inline css/properties/max-width.json]: webkit = true

File: css/properties/min-height.json [hunk 9]
```diff
@@ -163,6 +214,70 @@
...
+                {
+                  "version_added": "46"
+                },
+                {
+                  "prefix": "-webkit-",
+                  "version_added": "22"
+                }
+              ],
+              "chrome_android": {
+                "version_added": "46"
```
Comment (jpmedley) [inline css/properties/min-height.json]: webkit = 25

File: css/properties/min-height.json [hunk 10]
```diff
@@ -163,6 +214,70 @@
...
+                }
+              ],
+              "safari_ios": {
+                "version_added": null
+              },
+              "samsunginternet_android": {
+                "version_added": "5.0"
+              },
+              "webview_android": {
+                "version_added": "46"
```
Comment (jpmedley) [inline css/properties/min-height.json]: webkit = true

File: css/properties/min-width.json [hunk 11]
```diff
@@ -53,55 +53,140 @@
...
+                  "version_added": "46"
+                },
+                {
+                  "prefix": "-webkit-",
+                  "version_added": "22"
+                }
+              ],
               "chrome_android": {
-                "version_added": null
+                "version_added": "46"
```
Comment (jpmedley) [inline css/properties/min-width.json]: webkit = 25

File: css/properties/min-width.json [hunk 12]
```diff
@@ -53,55 +53,140 @@
...
+                }
+              ],
+              "safari_ios": {
+                "version_added": null
+              },
+              "samsunginternet_android": {
+                "version_added": "5.0"
+              },
+              "webview_android": {
+                "version_added": "46"
```
Comment (jpmedley) [inline css/properties/min-width.json]: webkit = true

File: css/properties/min-width.json [hunk 13]
```diff
@@ -53,55 +53,140 @@
...
+                {
+                  "version_added": "46"
+                },
+                {
+                  "alternative_name": "min-intrinsic",
+                  "version_added": "22"
+                }
+              ],
+              "chrome_android": {
+                "version_added": "46"
```
Comment (jpmedley) [inline css/properties/min-width.json]: webkit = 25

File: css/properties/min-width.json [hunk 14]
```diff
@@ -53,55 +53,140 @@
...
               "safari_ios": {
                 "version_added": null
               },
               "samsunginternet_android": {
-                "version_added": null
+                "version_added": "5.0"
               },
               "webview_android": {
-                "version_added": null
+                "version_added": "46"
```
Comment (jpmedley) [inline css/properties/min-width.json]: webkit = true

File: css/properties/min-width.json [hunk 15]
```diff
@@ -161,6 +246,72 @@
...
+                {
+                  "version_added": "46"
+                },
+                {
+                  "prefix": "-webkit-",
+                  "version_added": "22"
+                }
+              ],
+              "chrome_android": {
+                "version_added": "46"
```
Comment (jpmedley) [inline css/properties/min-width.json]: webkit = 25

File: css/properties/min-width.json [hunk 16]
```diff
@@ -161,6 +246,72 @@
...
+              ],
+              "safari_ios": {
+                "version_added": null
+              },
+              "samsunginternet_android": {
+                "alternative_name": "-webkit-fill-available",
+                "version_added": "5.0"
+              },
+              "webview_android": {
+                "version_added": "46"
```
Comment (jpmedley) [inline css/properties/min-width.json]: webkit = true

File: css/properties/height.json [hunk 17]
```diff
@@ -51,51 +51,48 @@
             "deprecated": false
           }
         },
-        "stretch": {
+        "max-content": {
```
Comment (jpmedley) [inline css/properties/height.json]: I'm not up to speed on these features. Has this file been alphabetized or is this a name change?

File: css/properties/max-height.json [hunk 18]
```diff
@@ -53,16 +53,15 @@
             "deprecated": false
           }
         },
-        "sizing_values": {
+        "max-content": {
```
Comment (jpmedley) [inline css/properties/max-height.json]: Has this file been alphabetized or is this a name change?

File: css/properties/max-width.json [hunk 19]
```diff
@@ -53,52 +53,137 @@
             "deprecated": false
           }
         },
-        "sizing_values": {
+        "max-content": {
```
Comment (jpmedley) [inline css/properties/max-width.json]: Has this file been alphabetized or is this a name change?

File: css/properties/min-height.json [hunk 20]
```diff
@@ -54,52 +54,103 @@
             "deprecated": false
           }
         },
-        "sizing_values": {
+        "max-content": {
```
Comment (jpmedley) [inline css/properties/min-height.json]: Has this file been alphabetized or is this a name change?

File: css/properties/min-width.json [hunk 21]
```diff
@@ -53,55 +53,140 @@
             "deprecated": false
           }
         },
-        "sizing_values": {
+        "max-content": {
```
Comment (jpmedley) [inline css/properties/min-width.json]: Has this file been alphabetized or is this a name change?

File: css/properties/max-height.json (hunk 4 above)
Comment (rachelandrew) [reply css/properties/max-height.json]: I'm afraid that I don't understand this comment as we don't use webkit: true anywhere, same as the other similar ones.

File: css/properties/height.json (hunk 17 above)
Comment (rachelandrew) [reply css/properties/height.json]: I've added these features to the file. I don't know why it's showing up like that.

File: css/properties/max-height.json (hunk 18 above)
Comment (rachelandrew) [reply css/properties/max-height.json]: I added the features to the file.

File: css/properties/max-width.json (hunk 19 above)
Comment (rachelandrew) [reply css/properties/max-width.json]: I added the features to the file

File: css/properties/block-size.json [hunk 22]
```diff
@@ -65,14 +65,169 @@
...
+            }
+          },
+          "status": {
+            "experimental": false,
+            "standard_track": true,
+            "deprecated": false
+          }
+        }
+      },
+      "max-content": {
```
[the remark this answers is not in the document]
Comment (rachelandrew) [reply css/properties/block-size.json]: I think this is now corrected, I have also added some changes based on @jpmedley's comments below but I didn't understand the change some of the comments were requesting. If I can figure out what to do I'll make the changes!

File: css/properties/height.json (hunk 17 above)
Comment (ddbeck) [reply css/properties/height.json]: Yeah I think this is just git being too clever about the diff—flipping between this file on `master` and this revision makes it a bit easier to follow. Resolving this.

File: css/properties/max-height.json (hunk 18 above)
Comment (ddbeck) [reply css/properties/max-height.json]: @jpmedley these features were decomposed (i.e., `sizing_values` was a combination of `fit-content`, `max-content`, and `min-content`). I suggest taking a look at the split diff instead of the unified diff when you see something like this

File: css/properties/max-height.json (hunk 3 above)
Comment (ddbeck) [reply css/properties/max-height.json]: @jpmedley could you be more specific about your suggestion here (and the one below for webview)? Should it look like the `chrome` entry right before this one (i.e., two support statements, one for prefixed version_added @ 25 and one unprefixed @ 46)? Or do you want one entry (webkit prefixed, version_added @ 25)?

File: css/properties/block-size.json (hunk 22 above)
[the remark this answers is not in the document]
Comment (ddbeck) [reply css/properties/block-size.json]: Thanks, @rachelandrew! I've tried to ask some follow up questions to Joe to help things along. Thanks again!

File: css/properties/max-height.json (hunk 4 above)
Comment (jpmedley) [reply css/properties/max-height.json]: I made a mistake here. Deleting the comment.

File: css/properties/max-height.json (hunk 3 above)
Comment (jpmedley) [reply css/properties/max-height.json]: This should match chrome exactly. I was listing what's missing.

File: css/properties/max-height.json (hunk 4 above)
Comment (jpmedley) [reply css/properties/max-height.json]: Rachel,

Sorry the confusion.

Webview became a Chromium fork starting with version 37. If for a particular feature, Chrome is 37 or greater, webview data should match it exactly. Because the webkit prefix for this feature has never been removed, there needs to be a webkit entry for webview. 

For items that landed in Chromium before 37, webview get's complicated. In the context of this database, webview is [defined to include non-Chromium webviews](https://github.com/mdn/browser-compat-data/blob/master/browsers/webview_android.json). But I don't have access to anything that tells me whether those non-Chromium webviews supported a particular property, much less what version it was in. 

With this particular property, I know that it was in Chromium webview in 37. Since I do not know whether this was in any other webview, `true` is the only thing that safely covers both cases.

Unless I missed something, this should cover all my webview comments. Ping me if I missed something.

Comment (rachelandrew) [reply css/properties/max-height.json]: I'm afraid I still don't understand how this should look. I understand the engine history, I don't understand what change you want made to this PR.

Comment (jpmedley) [reply css/properties/max-height.json]: It needs to look like this:
```
"webview_android": [
  {
    "version_added": "46"
  },
  {
    "version_added":"true",
    "prefix": "-webkit-:
  }
}
```

File: css/properties/max-height.json (hunk 3 above)
Comment (rachelandrew) [reply css/properties/max-height.json]: I believe I have got them all, but this PR has got somewhat messy.

File: css/properties/max-height.json (hunk 4 above)
Comment (rachelandrew) [reply css/properties/max-height.json]: thank you.

File: css/properties/max-width.json (hunk 5 above)
Comment (jpmedley) [reply css/properties/max-width.json]: This should match Chrome.

File: css/properties/max-width.json (hunk 6 above)
Comment (jpmedley) [reply css/properties/max-width.json]: This needs to be:

```
"webview_android": [
  {
      "version_added": "46"
  },
  {
    "prefix": "-webkit-",
    "version_added": "true"
  }
]
```

File: css/properties/max-width.json (hunk 7 above)
Comment (jpmedley) [reply css/properties/max-width.json]: This needs to match chrome.

File: css/properties/max-width.json (hunk 8 above)
Comment (jpmedley) [reply css/properties/max-width.json]:
```
"webview_android": [
  {
      "version_added": "46"
  },
  {
    "prefix": "-webkit-",
    "version_added": "true"
  }
]
```

File: css/properties/min-height.json (hunk 9 above)
Comment (jpmedley) [reply css/properties/min-height.json]: My earlier comment is an error. This should match chrome.

File: css/properties/min-height.json (hunk 10 above)
Comment (jpmedley) [reply css/properties/min-height.json]:
```
"webview_android": [
  {
      "version_added": "46"
  },
  {
    "prefix": "-webkit-",
    "version_added": "true"
  }
]
```

File: css/properties/min-width.json (hunk 11 above)
Comment (jpmedley) [reply css/properties/min-width.json]: This should match Chrome.

File: css/properties/min-width.json (hunk 12 above)
Comment (jpmedley) [reply css/properties/min-width.json]:
```
"webview_android": [
  {
      "version_added": "46"
  },
  {
    "prefix": "-webkit-",
    "version_added": "true"
  }
]
```

File: css/properties/min-width.json (hunk 13 above)
Comment (jpmedley) [reply css/properties/min-width.json]: Match Chrome.

File: css/properties/min-width.json (hunk 14 above)
Comment (jpmedley) [reply css/properties/min-width.json]:
```
"webview_android": [
  {
      "version_added": "46"
  },
  {
    "prefix": "-webkit-",
    "version_added": "true"
  }
]
```

File: css/properties/min-width.json (hunk 15 above)
Comment (jpmedley) [reply css/properties/min-width.json]: Match Chrome.

File: css/properties/min-width.json (hunk 16 above)
Comment (jpmedley) [reply css/properties/min-width.json]:
```
"webview_android": [
  {
      "version_added": "46"
  },
  {
    "prefix": "-webkit-",
    "version_added": "true"
  }
]
```

File: css/properties/max-width.json (hunk 6 above)
Comment (rachelandrew) [reply css/properties/max-width.json]: I believe this has already been done.

File: css/properties/max-width.json (hunk 5 above)
Comment (rachelandrew) [reply css/properties/max-width.json]: already done afaik

**Comment by rachelandrew** [comment]
I believe the changes have all been done although this has become an exercise in chasing my own tail as comments are being made while I'm trying to make updates on previous comments. I suggest that we merge this, and if there are still issues please raise them and I'll make a new PR to fix these. The aim of this was to support the Firefox release.
